physics/SFC_Models/Land/Noahmp/lnd_iau_mod.F90

```diff
@@ -0,0 +1,848 @@
+   Land_IAU_state%wt_normfact = 1.0
+   if (Land_IAU_Control%iau_filter_increments) then
+      ! compute increment filter weights, sum to obtain normalization factor
+      dtp=Land_IAU_Control%dtp
+      nstep = 0.5*Land_IAU_Control%iau_delthrs*3600/dtp
+      ! compute normalization factor for filter weights
+      normfact = 0.
+      do k=1,2*nstep+1
+         kstep = k-1-nstep
+         sx     = acos(-1.)*kstep/nstep
+         wx     = acos(-1.)*kstep/(nstep+1)
+         if (kstep .ne. 0) then
+            wt = sin(wx)/wx*sin(sx)/sx
+         else
+            wt = 1.0
+         endif
+         normfact = normfact + wt
+         if (Land_IAU_Control%me == Land_IAU_Control%mpi_root) print *,'filter wts',k,kstep,wt
+      enddo
+      Land_IAU_state%wt_normfact = (2*nstep+1)/normfact
+   endif
+
+   ! increment files in fv3 tiles 
+   if (trim(Land_IAU_Control%iau_inc_files(1)) .eq. '' .or. Land_IAU_Control%iaufhrs(1) .lt. 0) then ! only 1 file expected
+      print*, "warning! in Land IAU but increment file name is empty or iaufhrs(1) is negative"
+      return   
+   endif    
+   if (Land_IAU_Control%me == Land_IAU_Control%mpi_root) then
+      print *,"land_iau_init: Increment file ", trim(adjustl(Land_IAU_Control%iau_inc_files(1)))
+   endif
+
+   ! determine number of valid forecast hours
+!TODO: can read this from the increment file ("Time" dim)
+   ntimesall = size(Land_IAU_Control%iaufhrs)
+   ntimes = 0
+   do k=1,ntimesall
+      if (Land_IAU_Control%iaufhrs(k) .lt. 0) exit
+      if (Land_IAU_Control%me == Land_IAU_Control%mpi_root) then
+         print *,k, " fhour ", Land_IAU_Control%iaufhrs(k)
+      endif
+      ntimes = ntimes + 1
+   enddo
+   if (Land_IAU_Control%me == Land_IAU_Control%mpi_root) print *,'land_iau_init: ntimes = ',ntimes
+   Land_IAU_Control%ntimes = ntimes
+   if (ntimes < 1) then
+      return
+   endif
+   if (ntimes > 1) then
+      allocate(idt(ntimes-1))
+      idt = Land_IAU_Control%iaufhrs(2:ntimes)-Land_IAU_Control%iaufhrs(1:ntimes-1)
+      do k=1,ntimes-1
+         if (idt(k) .ne. Land_IAU_Control%iaufhrs(2)-Land_IAU_Control%iaufhrs(1)) then
+            print *,'in land_iau_init: forecast intervals in iaufhrs must be constant'
+         !   call mpp_error (FATAL,' forecast intervals in iaufhrs must be constant')
+            errmsg = 'Fatal error in land_iau_init. forecast intervals in iaufhrs must be constant'
+            errflg = 1
+            return
+         endif
+      enddo
+      deallocate(idt)
+   endif
+   dt = (Land_IAU_Control%iau_delthrs*3600.)
+   rdt = 1.0/dt
+   Land_IAU_state%rdt = rdt
+   if (Land_IAU_Control%me == Land_IAU_Control%mpi_root) print *,'land_iau interval, rdt',Land_IAU_Control%iau_delthrs,Land_IAU_state%rdt
+
+   ! Read all increment files at iau init time (at beginning of cycle) 
+   ! allocate (wk3_stc(n_t, 1:im,jbeg:jend, 1:km))  
+   call read_iau_forcing_fv3(Land_IAU_Control, errmsg, errflg)  !, wk3_stc, wk3_slc

```

Comment:
I didn't address this in https://github.com/tsga/ccpp-physics/pull/1, but the code isn't checking for errors after returning from `read_iau_forcing_fv3`. When I try to run the new test with the data missing, instead of exiting with a nice error message, you get a seg fault because this code just continues and tries to access variables that were never allocated when the data files are missing.